### PR TITLE
Reworked the blending gallery example so that mplcairo is not used

### PR DESCRIPTION
**Pull request by ayshih**
We have `mplcairo` as a docs dependency because of a single gallery example (#6835).  Unfortunately, it's been over two years since the last release of `mplcairo`, and there are no wheels for Python 3.11+.

This PR reworks the blending gallery example so that `mplcairo` is no longer used.  The blending operation has to be done "manually", but it's not too bad.  Accordingly, this PR also removes `mplcairo` as a dependency.

### CODE REVIEW

File: examples/plotting/screen_blend_mode.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+======================
+Blending maps together
+======================
+
+This example shows how to blend two maps.
+
+``matplotlib`` by itself provides only alpha-based transparency for
+superimposing one image onto another, which can be restrictive when trying to
+create visually appealing composite images.  For better results, one can
+use math to perform standard image-processing
```
Comment (Cadair) [inline examples/plotting/screen_blend_mode.py:11]: by the power of "math" :superhero:

Comment (Cadair) [inline examples/plotting/screen_blend_mode.py:11]:
```suggestion
use standard image-processing
```

?

**Review by Cadair — COMMENTED** [review]
lgtm

File: examples/plotting/screen_blend_mode.py (hunk 1 above)
Comment (ayshih) [reply examples/plotting/screen_blend_mode.py]: Now a bit reworded

**Review by Cadair — APPROVED** [review]

**Review by alasdairwilson — APPROVED** [review]

**Pull request merged**